### PR TITLE
Async Backing: Fix clippy errors in statement distribution branch

### PR DESCRIPTION
**Pull request by mrcnski**
Targets https://github.com/paritytech/polkadot/pull/6715.

I wanted to fix the failing test job too, but looks like there are failures from https://github.com/paritytech/polkadot/pull/5022. Which requires a fresh merge of `master`.

### CODE REVIEW

**Comment by slumber** [comment]
Can you please rebase it so that changes are displayed correctly?

**Comment by mrcnski** [comment]
@slumber Done.

**Review by slumber — APPROVED** [review]
Nice, thanks

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2401620

File: node/network/statement-distribution/src/vstaging/grid.rs [hunk 1]
```diff
@@ -60,10 +60,10 @@ pub struct SessionTopologyView {
 impl SessionTopologyView {
 	/// Returns an iterator over all validator indices from the group who are allowed to
 	/// send us manifests.
-	pub fn iter_group_senders<'a>(
-		&'a self,
+	pub fn iter_group_senders(
+		&self,
 		group: GroupIndex,
-	) -> impl Iterator<Item = ValidatorIndex> + 'a {
+	) -> impl Iterator<Item = ValidatorIndex> + '_ {
```
Comment (rphmeier) [inline node/network/statement-distribution/src/vstaging/grid.rs:66]: my grey beard is showing 😆

**Review by rphmeier — APPROVED** [review]

File: node/network/statement-distribution/src/vstaging/grid.rs (hunk 1 above)
Comment (mrcnski) [reply node/network/statement-distribution/src/vstaging/grid.rs:66]: Haha! I didn't even know this was a thing.

**Pull request merged**